<!doctype html>
<html lang="en">
  {% include 'head.html' %}

  <body style="position: relative;">
    {% include 'top_menu.html' %}

    <div class="container-fluid">
      <div class="row">
        {% include 'molecule/sidebar.html' %}

        <div class="col-10 offset-2 px-3 content">

          <h2 id="expression">The distribution of {{ molecule }} expression across TCGA cancers</h2>
          <div>
            <div id="expression-plot"></div>
          </div>
          {% if is_isomiR and at_least_one_highly_expressed %}
            <p>
              *{{ molecule }} is highly expressed in the corresponding cancer, i.e.,
              is included in the list of 5'-isomiRs contributing to 95% of total
              miRNA sequencing reads.
            </p>
          {% endif %}

          <h2 id="sequence-based-targets">Sequence-based predicted {{ "targets" if is_isomiR else "5'-isomiR regulators"}}</h2>
          <p>
            {% if is_isomiR %}
              Target genes of {{ molecule }} are predicted with the use of miRDB and TargetScan.
            {% else %}
              IsomiR regulators of {{ molecule }} are predicted with the use of miRDB and TargetScan.
            {% endif %}
            Only miRDB predictions with Target Score &geq; 80 are shown.
            The number of TargetScan predictions is harmonized with miRDB
            by selecting the same amount of targets with the lowest
            cumulative weighted context++ scores.
            Note that these predictions are based solely on isomiR and target gene sequences 
            (i.e., they are not cell/tissue specific).
          </p>
          <table class="table table-bordered table-striped">
            <thead>
              <tr>
                {% if is_isomiR %}
                  <th class="align-top">Target gene</th>
                {% else %}
                  <th class="align-top">5'-isomiR</th>
                {% endif %}
                <th class="align-top">miRDB Target Score</th>
                <th class="align-top">TargetScan cumulative weighted context++ score</th>
              </tr>
            </thead>
            <tbody>
              {% for target, row in targets_seq.iterrows() %}
                <tr>
                  <td style="width: 33.3%;">{{ target }}</td>
                  <td style="width: 33.3%;">{{ row['mirdb_score'] }}</td>
                  <td style="width: 33.3%;">{{ row['targetscan_score'] }}</td>
                </tr>
              {% endfor %}
            </tbody>
          </table>

          <h2 id="targets">Pan-cancer conserved {{ "targets" if is_isomiR else "5'-isomiR regulators"}}</h2>
          <p>
            To select
            {% if is_isomiR %}
              {{ molecule }} &#x2192; target
            {% else %}
              5'-isomiR &#x2192; {{ molecule }}
            {% endif %}
            interactions supported by expression data
            we calculate the number of TCGA projects
            where expression levels of 5'-isomiR and target gene
            are significantly anti-correlated (Spearman's correlation &lt; -0.3, <i>p</i> &lt; 0.05).
          </p>

          <ul>
          </ul>
          <table class="table table-bordered table-striped">
            <thead>
              <tr>
                <th class="align-top">Interaction</th>
                <th class="align-top">Number of cancers with significant anti-correlation</th>
                <th class="align-top">Cancers with significant anti-correlation</th>
              </tr>
            </thead>
            <tbody>
              {% for source, target, anti_corr_cancers in targets_conserved %}
                <tr>
                  <td style="width: 33.3%;">{{ source }} &#x2192; {{ target }}</td>
                  <td style="width: 33.3%;">{{ anti_corr_cancers|length}}</td>
                  <td style="width: 33.3%;">
                    {% for cancer in anti_corr_cancers %}
                      <a href="/cancer_isomir_target/{{ cancer.rstrip('*') }}/{{ source }}/{{ target }}">{{ cancer }}</a>{% if loop.index != anti_corr_cancers|length %}, {% endif %}
                    {% endfor %}
                  </td>
                </tr>
              {% endfor %}
            </tbody>
          </table>

          {% if not is_isomiR or (is_isomiR and at_least_one_highly_expressed) %}
            <p>
              *5'-isomiR is highly expressed in the corresponding cancer, i.e.,
              is included in the list of 5'-isomiRs contributing to 95% of total
              miRNA sequencing reads.
            </p>
          {% endif %}
          
          <h2 id="targets">See also</h2>
          <p>
            The detailed information on cancer-specific {{ molecule }} expression and {{ "targeting" if is_isomiR else "regulation by 5'-isomiRs"}}:
          </p>
          <ul>
              {% for cancer in cancers %} 
                <li><a href="/cancer_molecule/{{ cancer.rstrip('*') }}/{{ molecule }}">TCGA-{{ cancer }}</a></li>
              {% endfor %}
          </ul>
          {% if is_isomiR and at_least_one_highly_expressed %}
            <p>
              *{{ molecule }} is highly expressed in the corresponding cancer, i.e.,
              is included in the list of 5'-isomiRs contributing to 95% of total
              miRNA sequencing reads.
            </p>
          {% endif %}

        </div>

      </div>
    </div>
    
    {% include 'footer.html' %}
    <script src="{{ url_for('static', filename='lib/plotly-2.9.0-dist/js/plotly.min.js') }}"></script>
    <script>
      var molecule = "{{ molecule }}";
      {% if is_isomiR %}
        var units = "RPM";
      {% else %}
        var units = "TPM";
      {% endif %}
      var data = {{ expression|safe }};
    </script>    
    <script src="{{ url_for('static', filename='js/molecule.js') }}"></script>

  </body>
</html>
